### PR TITLE
Restored all folders / files.

### PR DESCRIPTION
**Pull request by abdurashidovtaha2**

### CODE REVIEW

File: oop/js/index.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+    <link rel="stylesheet" href="index.css">
+</head>
+
+<body id="root">
+    <form action="" id="task-form">
+        <input type="text" data-action="add-new-task" placeholder="Add your task">
+        <button type="submit">Create Task</button>
+    </form>
```
Comment (fayzzzm) [inline oop/js/index.html:15]: Can you generate this component dynamically as React Component? 
Example: `generateForm()` and it appends to body this form? It is a weird idea?

File: oop/js/index.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
```
Comment (fayzzzm) [inline oop/js/index.html:7]: Could you make this shit to be dynamically changeable? Could it show how many unfinished tasks do you have?

File: oop/js/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+const taskListEl = document.getElementById("tasks-list");
+
+class TaskList {
+    constructor() {
+        this.taskId = 1;
+        this.taskList = [];
```
Comment (fayzzzm) [inline oop/js/index.js:6]: If you want to have default `taskId`, `taskList`, just make it something like: `constructor(taskid = 1, taskList = [])`

File: oop/js/pages/main.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+taskFrm.onsubmit = evt => {
+    evt.preventDefault();
+
+    const value = newTaskInput.value;
+    
+    tasks.createTask(value);
+    tasks.renderList();
+    newTaskInput.value = ``;
+}
```
[the remark this answers is not in the document]
Comment (fayzzzm) [reply oop/js/pages/main.js:13]: Everything is good, I hope you can get my point of view, what I want to see in you. Keep going bro!!

File: oop/js/index.css [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#tasks-list li {
+    /* display: flex; */
+}
```
[the remark this answers is not in the document]
Comment (abdurashidovtaha2) [reply oop/js/index.css:3]: ye i'll try to)), though maybe not in this p

File: oop/js/index.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            taskEl.innerHTML = `
+            <p>${o.content}</p>
+            <button data-action="delete-task-${o.taskId}">X</button>
+            `;
+            taskListEl.appendChild(taskEl);
+
+            const deleteBtn = document.querySelector(`[data-action="delete-task-${o.taskId}"]`);
+            deleteBtn.onclick = () => {
+                this.deleteTask(o.taskId);
+            }
```
[the remark this answers is not in the document]
Comment (abdurashidovtaha2) [reply oop/js/index.js:27]: ye i got the idea, i'll try doing that

File: oop/js/index.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    createTask(content) {
+        this.taskList.push({ taskId: this.taskId++, content, taskComplete: false });
+        this.renderList();
+    }
+    deleteTask(taskId) {
+        this.taskList = this.taskList.map(o => o.taskId === taskId ? { ...o, taskComplete: true } : o);
+        this.renderList();
+    }
```
[the remark this answers is not in the document]
Comment (abdurashidovtaha2) [reply oop/js/index.js:38]: interesting, haven't thought about that, i'll try thought

File: oop/js/pages/main.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (abdurashidovtaha2) [reply oop/js/pages/main.js:13]: ye

**Pull request merged**